### PR TITLE
Add upstream.current_upstream_name() function to return the request's proxy target

### PR DESCRIPTION
**Pull request by pushrax**
Replaces https://github.com/pushrax/lua-nginx-module/pull/1

@PeiwenChen @csfrancis @Sirupsen for review

### CODE REVIEW

File: src/ngx_http_lua_upstream_module.c [hunk 1]
```diff
@@ -544,3 +548,43 @@ ngx_http_lua_upstream_find_upstream(lua_State *L, ngx_str_t *host)
...
 }
+
+
+static int
+ngx_http_lua_upstream_current_upstream_name(lua_State *L)
+{
+    ngx_http_request_t *r;
+    ngx_http_upstream_t *us;
+    ngx_http_upstream_conf_t *ucf;
+    ngx_http_upstream_srv_conf_t *uscf;
```
Comment (agentzh) [inline src/ngx_http_lua_upstream_module.c:559]: These variable names should be aligned up. Better use the `ngx-releng` tool to check your code style:

https://github.com/openresty/nginx-devel-utils

**Comment by agentzh** [comment]
Will you also add an entry to README too? Thanks!

File: src/ngx_http_lua_upstream_module.c (hunk 1 above)
Comment (pushrax) [inline src/ngx_http_lua_upstream_module.c:559]: Thanks for the tip. I'll start using that from now on.

**Comment by pushrax** [comment]
https://github.com/openresty/lua-upstream-nginx-module/pull/20

**Pull request closed without merging**